### DISCUSSION TITLE
Error loading workspace folders

### DISCUSSION THREAD

**Discussion by iamshreeram** [golang/vscode-go#1365] [Help]
Seeing **2 errors** while loading *vscode* from Go Extension

**Details of Error 1**
Go Extension is looking at all my project directories in the workspace and throws below error 

```
Error loading workspace folders (expected 11, got 9)
failed to load view for file:///Users/ram/project: exhausted
failed to load view for file:///Users/ram/project/node: exhausted
```

Below is my directory structure and entire `project` is added to workspace 

```
~
 | /.config
 | /.cabal/
 | /.nvm/
 | .bash_profile
 | .bashrc
 | /project/
 |  | /Clojure/
 |  |  | /allmycljprojects 
 |  | /Shell/
 |  |  | /allmybashprojects 
 |  | /rust/
 |  |  | /allmyrustprojects
 |  | /haskell/
 |  |  | /allmyhaskellprojects
 |  | /python/
 |  |  | /allmypythonprojects 
 |  | /go/
 |  |  | /bin
 |  |  | /pkg
 |  |  | /src
 |  |  |  | /github.com
 |  |  |  | /gitlab.com
 |  |  |  | /golang.org
 |  |  |  | /gopkg.in
 |  | /scala/
 |  |  | /allmyscalaprojects
 |  | /node/
 |  |  | /allmynodeprojects
```

`Go: Locate Configured Go Tools` is running `go env` on each directory of **workspace**. I want to explicitly specify the `golang` directory  that `go extension` should look only at. Is it possible?

**Details of Error 2**
I'm also seeing an error in another pop-up that says -
```
Error loading workspace: You are outside of a module and outside of $GOPATH/src. If you are using modules, please open your editor to a directory in your module. If you believe this warning is incorrect, please file an issue: https://github.com/golang/go/issues/new.
```

**gopls version** - `golang.org/x/tools/gopls v0.6.6`
**go version** - go1.15.2 darwin/amd64

**VSC details**
```

Version: 1.54.3
Commit: 2b9aebd5354a3629c3aba0a5f5df49f43d6689f8
Date: 2021-03-15T11:57:12.728Z (4 days ago)
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Darwin x64 19.6.0
```

Please let me know if any other details needed.

**Comment by iamshreeram** [comment]
I have fixed Error 2 by updating the **gopls** version to **0.6.9**. Error 1 still remains.

**Comment by stamblerre** [comment]
`gopls` doesn't currently support repositories with multiple modules by default. Please take a look at the documentation for [setting up a workspace](https://github.com/golang/tools/blob/master/gopls/doc/workspace.md) and see if one of the suggestions there addresses the issue.

**Reply by iamshreeram** [reply]
@stamblerre , Thanks for the help. `experimentalWorkspaceModule` is helping to remove the pop-up. But The same error is showing up when I hover on `package`.

On Error # 2:
I found a control `directoryFilters` that I would want to use to filter only **go directory** from my projects. (I want `gopls` to look at only the **go** directory in my project folder.)

Could you please help if below is right format?
```
    "gopls": {
         // "experimentalWorkspaceModule": true,
         "directoryFilters": ["+/~/project/go/"],
          ....
      }
```
Or Would I be able to use `"directoryFilters": ["-/Users/home/ram/project/node/","-node_modules"],` to make gopls to ignore all node directory?

Thanks!

**Reply by stamblerre** [reply]
Thanks for following up! I noticed that you also filed https://github.com/golang/go/issues/45171. Do you want to continue the discussion there so that we can resolve one issue at a time?

**Reply by iamshreeram** [reply]
@stamblerre , Yeah.  Sure. That would work.

Thanks!
